### DISCUSSION TITLE
Anyone want to talk about SLSA, GH Advisory DB, and DeWitt clauses?

### DISCUSSION THREAD

**Discussion by gps-iqt-labs** [Open-Source-Software-Neighborhood-Watch/town-square#14] [Q&A] [answered]
@metalcorebear's [introductory post](https://github.com/Open-Source-Software-Neighborhood-Watch/town-square/discussions/2#discussioncomment-2237505) notes:
> [GitHub’s Advisory Database](https://github.com/advisories) is now open to contributions from developers who might want to __share detailed information__  [emph. added] __about software vulnerabilities__, such as the versions that are affected.

Knowing @jspeed-meyers + others have [compelling views](https://blog.chainguard.dev/the-dirty-secret-of-cybersecurity-standards/) on the need for detailed vuln info, less opinionated (that is: more [evidence-based](https://blog.chainguard.dev/slsa-vs-software-supply-chain-attacks/)) cybersecurity standards, and greater fortitude in the face of DeWitt Clauses, I wanted to open up a thread here. Also, inviting @bentztozer.

A few questions you've all inspired me to ask...

- __Do you think platforms like GH's Advisory Database offer a viable way out of [the DeWitt problem](https://dwheeler.com/essays/dewitt-clause.html)?__ (At least where [community-developed software](https://books.google.com/books?id=oSsIfoDQHhgC&pg=PA109&lpg=PA109&dq=%22community-developed+software%22&source=bl&ots=Bc2-Tjcp-s&sig=ACfU3U06YBs0Xu8RL2sk5Zj61xE6J1XezA&hl=en&sa=X&ved=2ahUKEwiVpbGmn6P4AhUnj4kEHbeEB7wQ6AF6BAgpEAM#v=onepage&q=%22community-developed%20software%22&f=false) is concerned, accepting the premise of [It's Time to Take GitHub Threats Seriously](https://www.darkreading.com/application-security/it-s-time-to-take-github-threats-seriously), and acknowledging that [while it sure is active](https://www.howdoyou.dev/self-reported-attitudes-and-behaviors#q10-version-control), GitHub isn't the only game in town.) __In other words, would you characterize databases like these as a form of "[Vendor Truth Serum](https://cps-vo.org/node/35012)," however weak that serum?__ (At least for the contents of the [much-maligned `/vendor` directory](https://gist.github.com/datagrok/8577287).) Or do they not adequately "[answer the mail](https://twitter.com/bizbuzzies/status/643429174884106240?lang=en)" in your view, given the complexity/opacity of today's codebases, the [information-processing challenges](https://securityboulevard.com/2022/06/five-reasons-for-alert-fatigue-and-how-to-make-it-stop/) associated with these types of advisories, and/or the [legal dimensions](https://heinonline.org/HOL/LandingPage?handle=hein.journals/rol25&div=19&id=&page=) of the DeWitt problem (e.g., product disparagement litigation)?
- Knowing there are multiple competing/partially overlapping advisory databases, __how do you think about the associated [specialization vs. fragmentation](https://www.bloomberg.com/graphics/2015-paul-ford-what-is-code/#:~:text=The%20World%20Wide,love%20to%20party.) tradeoff (if any) in this context__?
- And, finally, knowing SLSA and SigStore are relevant to this discussion (but not being a SLSA expert myself :canned_food:), __are there advisory databases that SLSA does not currently incorporate, _but should work to integrate_ in your opinion?__ Or is this new paradigm more about ["providing farm-to-table guarantees for bits and bytes"](https://www.usenix.org/system/files/sec19-torres-arias.pdf) through cryptographically-signed packages, and less about threat intelligence/vuln info/CVE integration?

Genuinely curious as to what this community thinks...

**Comment by jspeed-meyers** [comment, marked as answer]
@gps-iqt-labs, all interesting!

On whether the GH advisory database answers the DeWitt mail: I don't think so. The DeWitt clause problem, in my view, is about the legal rights that software vendors can or cannot impose on their users and is unrelated to the GH advisory database. I personally think that software security products are under-benchmarked, which hurts anyone that relies on software, and that this under-benchmarking is partly the result of DeWitt clauses, which scare 3rd parties away from benchmarking security tools, including code security tools.

I'm fine with having lots of advisory databases. The bad ones won't get used, in the long run, and that's fine with me.

And SLSA is, you're right, more about farm-to-table guarantees and not about CVEs. It's really an "integrity" framework, which distinguishes it from the tools and frameworks that focus on code quality.

There's a lot here. I figure we can continue the conversation in person tonight!
